File: .py/Program/SSD.py
from Storage import Storage

class SSD (Storage):
    __readSpeed = 0
    __writeSpeed = 0

    def __init__ (self, brand = "", model = "", size = 0, readSpeed = 0, writeSpeed = 0):
        super ().__init__ (brand, model, size)
        self.__readSpeed = readSpeed
        self.__writeSpeed = writeSpeed

    def setReadSpeed (self, readSpeed):
        self.__readSpeed = readSpeed

    def setWriteSpeed (self, writeSpeed):
        self.__writeSpeed = writeSpeed

    def getReadSpeed (self):
        return self.__readSpeed
    
    def getWriteSpeed (self):
        return self.__writeSpeed
    
    def info (self):
        super ().info ()
        print (f"HDD { self.__readSpeed }MB/s { self.__writeSpeed }MB/s")

File: .cpp/Program/SSD.cpp
#pragma once
#include <bits/stdc++.h>
#include "Storage.cpp"
using namespace std;

class SSD : public Storage {
    private:
        int readSpeed;
        int writeSpeed;

    public:
        SSD () { }

        SSD (string brand, string model, int size, int readSpeed, int writeSpeed) : Storage (brand, model, size) {
            this->readSpeed = readSpeed;
            this->writeSpeed = writeSpeed;
        }

        void setReadSpeed (int readSpeed) { this->readSpeed = readSpeed; }
        void setWriteSpeed (int writeSpeed) { this->writeSpeed = writeSpeed; }

        int getReadSpeed () { return readSpeed; }
        int getWriteSpeed () { return writeSpeed; }

        void info () override { Storage::info(); cout << " SSD " << readSpeed << "MB/s " << writeSpeed << "MB/s" << endl; }

        ~SSD () { }
};